src/components/Settings/SettingsPlex.tsx

```diff
@@ -52,12 +62,143 @@ interface SyncStatus {
   libraries: Library[];
 }
 
+interface PresetServerDisplay {
+  name: string;
+  ssl: boolean;
+  uri: string;
+  address: string;
+  host: string;
+  port: number;
+  local: boolean;
+}
 interface SettingsPlexProps {
   onComplete?: () => void;
 }
 
+function PlexPresetOption(props: { key: string; server: PresetServerDisplay }) {
+  if (props.server.local) {
+    return (
+      <option key={props.server.uri} value={props.server.uri}>
+        {' '}
+        {props.server.name}
+        {' ('}
+        {props.server.address}
+        {') [local]'}
+      </option>
+    );
+  }
+  return (
+    <option key={props.key} value={props.server.uri}>
+      {' '}
+      {props.server.name}
+      {' ('}
+      {props.server.address}
+      {') [remote]'}
+    </option>
+  );
+}
+
 const SettingsPlex: React.FC<SettingsPlexProps> = ({ onComplete }) => {
   const intl = useIntl();
+  const { addToast, removeToast } = useToasts();
+  const availablePresets = (): Map<string, PresetServerDisplay> => {
+    const map: Map<string, PresetServerDisplay> = new Map<
+      string,
+      PresetServerDisplay
+    >();
+    // turns the URI into a usable hostname / IP
+    const regexp = /(http(s?):\/\/)(.*)(:[0-9]*)/;
+    dataServers?.forEach((dev) => {
+      if (dev.provides.includes('server') && dev.owned == true) {
+        dev.connection.forEach((conn) => {
+          const theHost = conn.uri.replace(regexp, '$3');
+          map.set(conn.uri, {
+            name: dev.name,
+            ssl: conn.protocol == 'https' ? true : false,
+            uri: conn.uri,
+            address: conn.address,
+            port: conn.port,
+            local: conn.local,
+            host: theHost,
+          });
+        });
+      }
+    });
+    return map;
+  };
+  const presetServerSelected = (): string => {
+    let resp = 'manual';
+    if (isPresetServer) {
+      availablePresets().forEach((srv) => {
+        if (
+          srv.host == data?.ip &&
+          srv.port == data.port &&
+          srv.ssl == data?.useSsl
+        ) {
+          resp = srv.uri;
+        }
+      });
+    }
+    return resp;
+  };
+  const changePresetServer = async (
+    e: React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    setIsUpdatingPreset(true);
+    const targPreset = availablePresets().get(e.target.value);
+    if (targPreset) {
+      setIsPresetServer(true);
+      let toastID: string | undefined = undefined;
```

Comment:
By not assigning anything, it's undefined.

```suggestion
      let toastID: string | undefined;
```